ls1intum/Artemis
# [Feature] Show not rated results in Exercise Details

### Checklist

### Motivation and Context


This PR aims to close issue #2732. Especially for programming exercises, we want students to be able to practice an exercise and continue working on an exercise even after the due date has passed.

### Description

When the due date of an prog. ex. has passed, the student is able to still submit and push his code changes and get (automatic feedback). But the result is not graded and doesn't change the received grade. 
All results (submissions) can be viewed in the Exercise Details of an exercise and are shown as a timeline. 
The issue was that only graded results were displayed. This led to the issue, that if a student only had not rated results (started the participation after the due date), the history wasn't shown.


### Steps for Testing


1. Log in to Artemis
2. Create a programming exercise and set only the due date
3. Participate as a student before the due date and submit at least twice. This enables the result timeline to trigger.
4. Go to the Exercise Details page and verify that all your results are shown on the bottom of the side. Also verify that the latest (rated) result is displayed at the top of the page.
5. Now wait for the due date to pass and submit new code changes. Verify that the result in the Course Overview (exercise row) and Exercise Detail at the top are the last `rated` results. The 'not rated' result is only shown in the online editor and in the result history.
6. Login as a new student and participate in the same exercise (now after the due date) 
7. Submit new code and verify that the result in the Course Overview (exercise row) and Exercise Detail at the top are marked as `No graded result`. Also at the bottom of Exercise Details verify that the just created `not rated` result is displayed.
8. After at least two submissions, the result history should show at the bottom of Exercise Details.

### Test Coverage



Stays the same. 

### Screenshots



## Started exercise before due date:
### Code Editor showing ungraded result:
![image](https://user-images.githubusercontent.com/33484318/107206684-a6be5300-69ff-11eb-863b-8f1b41dfecf9.png)

### Exercise Details showing result history with graded and ungraded result:
![image](https://user-images.githubusercontent.com/33484318/107206741-bb025000-69ff-11eb-8095-66816786c44f.png)

### Course Exercise Row is still showing the lastest rated result:
![image](https://user-images.githubusercontent.com/33484318/107208672-382ec480-6a02-11eb-882b-d7522bd14ae8.png)

## Started exercise after the due date:
### Exercise details is always showing that no graded result is available:
![image](https://user-images.githubusercontent.com/33484318/107207487-a4102d80-6a00-11eb-9c29-1addfd3966dc.png)

### Course Exercise Row is always showing that no graded result is available:
![image](https://user-images.githubusercontent.com/33484318/107207838-1254f000-6a01-11eb-89d2-1c1e1fd91c0d.png)

### First ungraded result shows no history:
![image](https://user-images.githubusercontent.com/33484318/107207245-5c89a180-6a00-11eb-8514-5c00e1aad197.png)

### Consecutive results are shown as history:
![image](https://user-images.githubusercontent.com/33484318/107207543-b38f7680-6a00-11eb-9ea8-87d615e0062a.png)
